core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -164,6 +164,7 @@ class KafkaApis(val requestChannel: RequestChannel,
     val response =
       if (authorize(request.session, ClusterAction, Resource.ClusterResource)) {
         val (result, error) = replicaManager.stopReplicas(stopReplicaRequest)
+        stopReplicaRequest.partitions.asScala.foreach(topicPartition => coordinator.groupManager.removeOffsetsByTopicPartition(topicPartition))
```

Comment:
I'm not sure, but this doesn't seem right to me. The stop replica request will only be received by the brokers which own the deleted topic, but the partition leaders of the consumer offsets topic might be another set of brokers. Would it make more sense to hook this into metadata updates when we notice that a topic has been removed?